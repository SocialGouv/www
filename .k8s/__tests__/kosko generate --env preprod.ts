//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --preprod", async () => {
  process.env.RANCHER_PROJECT_ID = "c-kk8xm:p-s5gkl";
  process.env.HARBOR_PROJECT = "fabrique";
  expect(
    await getEnvManifests("preprod", "", project("www").preprod)
  ).toMatchSnapshot();
});
