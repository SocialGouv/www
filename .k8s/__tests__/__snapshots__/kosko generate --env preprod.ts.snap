// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --preprod 1`] = `
"---
metadata:
  annotations:
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: 'c-kk8xm:p-s5gkl'
    git/branch: v1.2.3
    git/remote: >-
      https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/www.git
    app.gitlab.com/app: socialgouv-www
    app.gitlab.com/env: preprod-dev2
    app.gitlab.com/env.name: preprod-dev2
  labels:
    azure-pg-admin-user: www
    application: v1-2-3-www
    owner: www
    team: www
    cert: wildcard
  name: www-85-preprod-dev2
apiVersion: v1
kind: Namespace
---
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-www
    app.gitlab.com/env: preprod-dev2
    app.gitlab.com/env.name: preprod-dev2
  labels:
    app: app
    application: v1-2-3-www
    owner: www
    team: www
    cert: wildcard
    component: nginx
  name: app
  namespace: www-85-preprod-dev2
spec:
  replicas: 1
  selector:
    matchLabels:
      app: app
  template:
    metadata:
      annotations:
        app.gitlab.com/app: socialgouv-www
        app.gitlab.com/env: preprod-dev2
        app.gitlab.com/env.name: preprod-dev2
      labels:
        app: app
        application: v1-2-3-www
        owner: www
        team: www
        cert: wildcard
        component: nginx
    spec:
      containers:
        - image: 'registry.gitlab.factory.social.gouv.fr/socialgouv/www/app:1.2.3'
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: app
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 32Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /index.html
              port: http
            periodSeconds: 5
apiVersion: apps/v1
kind: Deployment
---
metadata:
  labels:
    app: app
    application: v1-2-3-www
    owner: www
    team: www
    cert: wildcard
  name: app
  annotations:
    app.gitlab.com/app: socialgouv-www
    app.gitlab.com/env: preprod-dev2
    app.gitlab.com/env.name: preprod-dev2
  namespace: www-85-preprod-dev2
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: app
  type: ClusterIP
apiVersion: v1
kind: Service
---
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    app.gitlab.com/app: socialgouv-www
    app.gitlab.com/env: preprod-dev2
    app.gitlab.com/env.name: preprod-dev2
  labels:
    app: app
    application: v1-2-3-www
    owner: www
    team: www
    cert: wildcard
  name: app
  namespace: www-85-preprod-dev2
spec:
  rules:
    - host: preprod-www.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              serviceName: app
              servicePort: 80
            path: /
  tls:
    - hosts:
        - preprod-www.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
apiVersion: extensions/v1beta1
kind: Ingress
"
`;
